jmsSource/README.md

```diff
@@ -0,0 +1,343 @@
+
+### Options Available for sender
+**NOTE**: sender section is REQUIRED, but the following values are optional.
+
+*   **queues**: Optional. A list of queues for which the connector will be configured to send messages.
+*   **topics**: Optional. A list of topics for which the connector will be configured to send messages.
+*   **messageHandler**: Optional. See [Adding Message Handlers](#msgHandlers). A section containing two sub-sections: *queues* 
+and *topics*. These sub-sections should contain the name of a given queue or topic as a field, and the fully qualified class 
+name of a custom Message Handler as the value. Creating a custom Message Handler is explained later in this document. This is 
+completely optional, and if no custom message handlers are specified, a message handler that can handle the generic JMS 
+messages will be used.
+
+### Options Available for receiver
+**NOTE**: receiver section is REQUIRED, but the following values are optional.
+
+*   **queues**: Optional. A list of the queues that will be configured to receive messages. Messages from this list of
+queues will only be read by querying the source using a SELECT statement, which is explained [later in this document](#select).
+*   **queueListeners**: Optional. A list of the queues that will be configured to receive messages. Messages from this 
+list of queues will be read using a Message Listener, which will send the messages back to the source as a notification. Messages from these queues will *not* be available for query from SELECT statements.
+*   **topics**: Optional. A list of the topics that will be configured to receive messages. Messages from topics will be read 
+using a Message Listener, which will send the messages back to the source as a notification.
+*   **messageHandler**: Optional. [See Adding Message Handlers](#msgHandlers). A section containing three sub-sections: 
+*queues*, *queueListeners* and *topics*. These sub-sections should contain the name of a given queue or topic as a field, and 
+the fully qualified class name of a custom Message Handler as the value. Creating a custom Message Handler is explained later 
+in this document. This is completely optional, and if no custom message handlers are specified, a message handler that can 
+handle the generic JMS messages will be used.
+
+## Messages from the Source
+
+Messages that are sent from the JMS Server to a source as VANTIQ Notifications, (from either a queueListener or a topic) are 
+JSON objects in the following format:
+```
+{
+    JMSMessage:<theReceivedMessage>, 
+    JMSFormat:<theJMSMessageFormat>, 
+    JMSDestination:<queueName/topicName>
+}
+```
+
+The following example shows a rule that could be used to process incoming notifications to a JMS Source named JMS1:
+
+```
+RULE jmsMessageListener
+
+WHEN MESSAGE ARRIVES FROM JMS1 as msg
+var myObj = {}
+myObj.JMSMessage = msg.message
+myObj.JMSFormat = msg.JMSFormat
+myObj.JMSDestination = msg.destination
```

Comment:
What about the other infrastructure JMS props?  Things like JMSMessageId, correlationId, replyTo, deliveryMode, expiration, timestamp, priority, redelivered?  Seems like we should at least return these in the map.

Seems like we should probably allow them to be set as well, where appropriate.